src/logic/sync/synchronizer.ts

```diff
@@ -0,0 +1,194 @@
+    { abortSignal, interval }: { abortSignal: AbortSignal; interval: number }
+  ): Promise<void> {
+    while (running && !abortSignal.aborted) {
+      logger.info('Starting sync profiles loop')
+      await callToLoop(abortSignal).catch((error) => {
+        logger.error('Error in loop', { error: error.message })
+      })
+      if (!abortSignal.aborted && running) {
+        // Create a cancellable timeout that respects the abort signal
+        await new Promise<void>((resolve) => {
+          const timeoutId = setTimeout(() => resolve(), interval)
+          abortSignal.addEventListener('abort', () => {
+            clearTimeout(timeoutId)
+            resolve()
+          })
+        })
+      }
+    }
+    if (abortSignal.aborted) {
+      logger.info('Loop aborted')
+    }
+  }
+
+  async function syncProfiles(fromTimestamp: number, abortSignal: AbortSignal): Promise<void> {
+    if (syncWorkflowRunning) {
+      logger.warn('Sync workflow already running, skipping concurrent call')
+      return
+    }
+    syncWorkflowRunning = true
+    try {
+      const isAWeekOld = Date.now() - fromTimestamp > ONE_DAY_MS * 7
+      let cursor: number = fromTimestamp
+
+      // Process snapshots first if needed, and wait for completion before starting pointer-changes loop
+      if (isAWeekOld && !abortSignal.aborted) {
+        logger.info('Processing snapshots before starting pointer-changes loop')
+        cursor =
+          (await withRetry(async () => await snapshotsHandler.syncProfiles(fromTimestamp, abortSignal))) ?? cursor
+        cursor && (await memoryStorage.set(SYNC_STATE_KEY, cursor))
+        logger.info('Snapshots processing completed, starting pointer-changes loop', { cursor })
+      }
+
+      const pointerChangesLoopPromise = loop(
+        async (signal) => {
+          const newCursor = await pointerChangesHandler.syncProfiles(cursor, signal)
+          if (newCursor && newCursor > cursor) {
+            cursor = newCursor
+            await memoryStorage.set(SYNC_STATE_KEY, cursor)
+          }
+        },
+        { abortSignal, interval: POINTER_CHANGES_POLL_INTERVAL_MS }
+      )
+
+      const retrierLoopPromise = loop(
+        async (signal) => {
+          await failedProfilesRetrier.retryFailedProfiles(signal)
+        },
+        { abortSignal, interval: FAILED_PROFILES_RETRY_INTERVAL_MS }
+      )
+
+      // Track promises for monitoring - handled by Promise.all below
+      // eslint-disable-next-line @typescript-eslint/no-floating-promises
+      loopPromises = [pointerChangesLoopPromise, retrierLoopPromise]
+
+      void Promise.all(loopPromises)
+    } finally {
+      syncWorkflowRunning = false
+    }
+  }
+
+  async function start(_startOptions?: any): Promise<void> {
+    const shouldStartSynchronizer = (await config.getString('DISABLE_PROFILE_SYNC')) === 'true'
+    if (shouldStartSynchronizer) {
+      logger.info('Profile sync is disabled, skipping')
+      return
+    }
```

Comment:
Can we rename the const here, it sounds confusing. Maybe `isSinchronizerDisabled` or similar.